### ISSUE TITLE
Cannot use references for path params

### ISSUE THREAD

**Issue by fabiohecht** [openapi4j/openapi4j#131]
**Version**  
1.0.3  

**Describe the bug**  
From what I understand, something like this is valid according to the spec:

```
(...)
paths:
  /path/{pathParam}:
    parameters:
      - $ref: '#/components/parameters/path-pathParam'
    get:
      responses:
        200:
          description: All is fine
components:
  parameters:
    path-pathParam:
      name: pathParam
      in: path
      description: The path param
      required: true
      schema:
        type: string
```

**Expected behavior**  
I'm not sure I'm doing something wrong, if it's a bug, or feature request, but I expected this to work.

Instead, I get:
```
java.lang.RuntimeException: OpenApi3 validation failure
Validation error(s) :
paths./path/{pathParam}.required: Parameter 'null' in path '/path/{pathParam}' must have 'required' property set to true (code: 122)
From: paths./path/{pathParam}.required
paths./path/{pathParam}.name: Path parameter 'pathParam' in path '/path/{pathParam}' is expected but undefined (code: 124)
From: paths./path/{pathParam}.name
```

**Comment by llfbandit** [comment]
Changes will be available soon in 1.0.4-SNAPSHOT.

**Comment by fabiohecht** [comment]
great to know, thanks!
